src/fmt.ml

```diff
@@ -65,6 +70,10 @@ let rec list ?(pp_sep = cut) pp_v ppf = function
 | v :: vs ->
     pp_v ppf v; if vs <> [] then (pp_sep ppf (); list ~pp_sep pp_v ppf vs)
 
+let hashtbl ?(pp_sep = cut) ~pp_k ~pp_v ppf tbl =
+  let f ppf k v = pf ppf "@[<1>%a:@.%a@]%a" pp_k k pp_v v pp_sep () in
+  Hashtbl.iter (f ppf) tbl
```

Comment:
I'm not a big user of hashtables so don't have strong opinions on how they should be printed but maybe we should try to find commonalities or a pattern with [`Astring.String.Map.pp`](https://github.com/dbuenzli/astring/blob/8856c13eee4e4b4a23f1f083ba9c73efa30e9ea2/src/astring_string.ml#L589-L598). Also is there any reason why you don't enclose it in a box (I never remember which is best, leave it to the client or do it in the data structure pp itself).